app/src/main/java/com/spascoding/taskycourse/feature_auth/presentation/login/LoginScreen.kt

```diff
@@ -37,12 +40,23 @@ import com.spascoding.taskycourse.feature_auth.presentation.components.PasswordO
 import com.spascoding.taskycourse.feature_auth.presentation.util.AuthPattern
 import com.spascoding.taskycourse.navigation.Navigation
 import com.spascoding.taskycourse.ui.theme.TaskyCourseTheme
+import kotlinx.coroutines.flow.collectLatest
 
 @Composable
 fun LoginScreenRoot(
     navController: NavController,
     viewModel: LoginViewModel = hiltViewModel(),
 ) {
+    val context = LocalContext.current
+    LaunchedEffect(Unit) {
+        viewModel.toastMessages.collectLatest { userEvent ->
+            val errorMessage = when (userEvent) {
+                is LoginViewModel.UserEvent.Error -> userEvent.error.asString(context)
+            }
+            Toast.makeText(context, errorMessage, Toast.LENGTH_SHORT).show()
+        }
+    }
```

Comment:
Strictly speaking, you need something like this to collect one-time events 100% safely without risking to lose some:
```
@Composable
private fun <T> ObserveAsEvents(flow: Flow<T>, onEvent: (T) -> Unit) {
    val lifecycleOwner = LocalLifecycleOwner.current
    LaunchedEffect(flow, lifecycleOwner.lifecycle) {
        lifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
            withContext(Dispatchers.Main.immediate) {
                flow.collect(onEvent)
            }
        }
    }
}
```
For more info, I talk about the problem in this video: https://youtu.be/njchj9d_Lf8?si=0to4KwRHzEIECID8